bindings/wasm/src/did/wasm_document.rs

```diff
@@ -139,20 +238,84 @@ impl WasmDocument {
   ///
   /// Throws an error if the method is not found.
   #[wasm_bindgen(js_name = resolveMethod)]
-  pub fn resolve_method(&self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
+  pub fn resolve_method(&self, query: &UDIDUrlQuery, scope: Option<WasmMethodScope>) -> Result<WasmVerificationMethod> {
     let method_query: String = query.into_serde().wasm_result()?;
+    if let Some(scope) = scope {
+      return Ok(WasmVerificationMethod(
+        self
+          .0
+          .resolve_method_with_scope(&method_query, scope.0)
+          .ok_or(identity::did::Error::MethodNotFound)
+          .wasm_result()?
+          .clone(),
+      ));
+    }
     Ok(WasmVerificationMethod(
       self.0.try_resolve_method(&method_query).wasm_result()?.clone(),
     ))
   }
 
+  /// Attempts to resolve the given method query into a method capable of signing a document update.
+  #[wasm_bindgen(js_name = resolveSigningMethod)]
+  pub fn resolve_signing_method(&mut self, query: &UDIDUrlQuery) -> Result<WasmVerificationMethod> {
```

Comment:
Discussion: another place where we should consider returning `Option` rather than erroring? This would affect a lot of methods, so I'm okay leaving everything as fallible in this PR, to be clear.